File: project/src/main/webapp/resources/js/report.js
function report() {
    var boardUserCode = document.querySelector('.boardUsercode').value;
    var userCode = document.querySelector('.userCode').value;
    var boardCode = document.querySelector('.boardCode').value;

    var reportingUserCode = $("#regUserCode").val() ? $("#regUserCode").val() : null;
    var reportContent = $("#reportContent").val() ? $("#reportContent").val() : null;

    var report = {
        reportCode: null,
        boardCode: boardCode,
        reportedUserCode: boardUserCode,
        reportingUserCode: reportingUserCode,
        reportDate: null,
        reportContent: reportContent
    };

    console.log(report);

    var reportString = JSON.stringify(report);
    console.log(reportString);

    $.ajax({
        type: "post",
        url: "/project/board/free/report/" + boardCode,
        data: reportString,
        contentType: "application/json",

        success: function(response) {
            alert("신고되었습니다.");
            window.location.href = response.redirect;
        },
        error: function(xhr, status, error) {
            console.log("오류");
        }
    });
}

File: project/src/main/java/com/acorn/project/report/domain/Report.java
package com.acorn.project.report.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Report {
	String reportCode;
	String boardCode;
	String reportingUserCode;
	String reportedUserCode;
	String reportDate;
	String reportContent;
}

//reportCode varchar(255) primary key,
//boardCode VARCHAR(255) NOT NULL,
//reportingUserCode VARCHAR(255) NOT NULL,
//reportedUserCode VARCHAR(255) NOT NULL,
//FOREIGN KEY (reportedUserCode)
//REFERENCES usertbl(userCode) ON UPDATE CASCADE,
//reportDate datetime,
//reportContent varchar(255)
